src/services/getEditsForFileRename.ts

```diff
@@ -151,7 +151,8 @@ namespace ts {
                     const toImport = oldFromNew !== undefined
                         // If we're at the new location (file was already renamed), need to redo module resolution starting from the old location.
                         // TODO:GH#18217
-                        ? getSourceFileToImportFromResolved(resolveModuleName(importLiteral.text, oldImportFromPath, program.getCompilerOptions(), host as ModuleResolutionHost), oldToNew)
+                        ? getSourceFileToImportFromResolved(resolveModuleName(importLiteral.text, oldImportFromPath, program.getCompilerOptions(), host as ModuleResolutionHost),

```

Comment:
instead of passing array of paths pass in files themselves so that we don't need to allocate array everytime. 
You can use `find or contains` instead on array to do the search

Reply:
Done.